### PR TITLE
Add getShallowDiffs utility

### PR DESCRIPTION
**Pull request by ItsJonQ**
## Add getShallowDiffs utility

This update adds getShallowDiffs, a utility to help spot the differences between
prev/next props. This is handy when trying to figure out why a React component
has updated, by tapping into lifecycle hooks, such as `componentDidUpdate`.

### CODE REVIEW

File: src/getShallowDiffs.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * @param {Object} next nextProps/nextState
+ * @returns {Object} The differences
+ */
+function getShallowDiffs(previous: Object, next: Object): Object {
+  const diffs = Object.keys(previous).reduce((diffs, key) => {
+    if (previous[key] !== next[key]) {
+      return [...diffs, key]
+    }
+    return diffs
+  }, [])
```
Comment (knicklabs) [inline src/getShallowDiffs.js]: Couldn't we just use the simpler `.filter` method here instead? And shouldn't the return type be an array? Perhaps I'm missing something, but isn't this functionally equivalent to:

```
function getShallowDiffs(previous: Object, next: Object): Array {
  return Object.keys(previous).filter(key => key !== next[key])
}
```

Comment (ItsJonQ) [reply src/getShallowDiffs.js]: Oops! I merged it before I saw your question! sorry :D.

The output I have is this:

```
{
  diffs: ['propName'],
  previous: { propName: 'a' },
  next: { propName: 'b' },
}
```

From using it a bit.. I found that this structure was the most helpful. Hope that makes sense!

Comment (knicklabs) [reply src/getShallowDiffs.js]: Yeah, so the return type of `getShallowDiffs` should be an array. And I think the filter approach would be more concise.

Comment (ItsJonQ) [reply src/getShallowDiffs.js]: Ooops!! I misread. My bad 😅 

You are correct! I'll make the update

**Pull request merged**